### DISCUSSION TITLE
[Question] Would it be possible to use Coc.nvim for completion but other plugins for the rest? (mainly native Nvim LSP)

### DISCUSSION THREAD

**Discussion by pocco81** [neoclide/coc.nvim#3115] [Q&A] [answered]
Hello! Sorry if my question is "against" the purpose of Coc.nvim. I recently migrated from COC to the Native Nvim LSP and other plugins for the rest (completion, linting, etc). I really love Coc's completion and was wondering if it was possible to **only use** that feature and other plugins for the rest.

If so, how would I set it up? (Again, sorry if I'm asking for a lot, a simple yes/no to the main question is just as enough)

Thanks for the amazing work!

**Comment by fannheyward** [comment]
https://www.reddit.com/r/neovim/comments/nmcxjn/is_it_possible_to_use_cocnvim_only_for_completion/

It's not recommended to use the two at same time because there will be two LS run for your current buffer, this may cause some other problems. Try at your own risk.

**Comment by chemzqm** [comment, marked as answer]
You can write a coc.nvim extension which register completion source using other vim plugin's completion implementation, the API is https://github.com/neoclide/coc.nvim/blob/master/typings/index.d.ts#L4470
